### ISSUE TITLE
Wifiphisher has no such thing as a "log", is far from portable and spits out files everywhere!

### ISSUE THREAD

**Issue by yungtravla** [wifiphisher/wifiphisher#704]
This tool is amazing, but super frustrating, messy, and lacking essential stuff in so many areas...

First of all, no offense to anyone who has worked on this, but I can't call the following anything other than an epic fail:

**The logs are split in 3 sections (which completely messes up the order), and each section is limited to 5 lines (wtf)**

**1.** This is not OK, and it doesn't look better/neater, it's just confusing and frustrating. We need **all** the logs **in the right order**, so that, you know, it actually works as a log instead of weird, semi-useful, data accumulating gauges... And I can't forget to mention the fact that all logs get cleared on exit, but who cares, right? They were just super valuable logs with crucial information...

**2.** Setup.py (build & install) is completely redundant and it litters shit all over the f***ing place... At least include a decent uninstall option if that's what's gonna happen. This is mainly why I get so frustrated with this program. Again, no offense to anyone.

**3.** Wifiphisher should be able to work 'right out of the box', no build or installation **needed**, just run it, no matter where it's run from. That's a really easy thing to do, just set a variable with the current working directory on each session!

This project started end 2014, and in my opinion some of the most important features are still left out/half developed. It's advertised beautifully, and has gained much attention, but I have to say it's being maintained poorly, which forces me to resort to manual methods/countless modifications.

My apologies if I come across as arrogant or offensive, but I am really frustrated over these super obvious things.

I mean, would you buy a 4x4 ute just so you can use the airconditioning?

**Comment by anakin1028** [comment]
@yungtravla thanks for the suggestions

These are my opinions and maybe @blackHatMonkey and @sophron have other ideas and input:

**1** Indeed logging is important for our tool, and I think this will be implemented in the future.
**3**  Wifiphisher is a complex stuff in my mind since it needs to implement lots of security attacks (like KARMA attack). In order to provide a better python binding we need to patch the `hostapd` by ourself and this is a reason why currently we need to build and install. But I think portability is important, I guess we may come out a light version of wifiphisher in the future.
**4** Yes. This is a good idea.

Thanks : )

**Comment by blackHatMonkey** [comment]
@yungtravla I understand you might be frustrated with the tool but to call it an epic fail is too much. The core team has put in thousands of hours each of their free time since the start. If you actually did the comparison with the original and what it is now you can defiantly see the difference. Back when the project was started it only had a 1000 LOC but we are now at 8000 LOC with many bugs fixed and new ones introduced so progress has been  made.  Debates have been made about how do we go about logging but nothing concrete. With a growing platform it is going to be harder to implement features since multiple angles has to be considered and there is going to be conflicts. Over time this feature was not made a priority and not worked at all if any as you can't tell people what to work on.  You seem to have good ideas but your points could be better made with a different tone :smile:

**Comment by yungtravla** [comment]
>  Debates have been made about how do we go about logging but nothing concrete. With a growing platform it is going to be harder to implement features since multiple angles has to be considered and there is going to be conflicts.

That's right, I forgot to say that it isn't modular either. And I've read many, many discussions on "logs clearing on exit, limited to 5 lines only, etc.".

I also said I wasn't trying to come across that way, but this tool has been around for a long time, is still messy and lacking even a simple logging feature.

Nonetheless, I like this tool and appreciate all the effort that has been put into it, but I find it hard to ignore the fact that so much time was spent on trivial things and enhancements rather than core hacking features.

**Comment by sophron** [comment]
@yungtravla It seems to me that you are not aware of how the open-source model works. I recommend you to read <a href="http://www.catb.org/esr/writings/cathedral-bazaar/cathedral-bazaar/">The Cathedral and the Bazaar</a>.

Long story short, instead of complaining about the software, help make it better (a good way is by writing code).

> 1. This is not OK, and it doesn't look better/neater, it's just confusing and frustrating. We need all the logs in the right order, so that, you know, it actually works as a log instead of weird, semi-useful, data accumulating gauges... And I can't forget to mention the fact that all logs get cleared on exit, but who cares, right? They were just super valuable logs with crucial information...

I think you are confusing logs with the output on the screen. Read (again?) my answer <a href="https://github.com/wifiphisher/wifiphisher/issues/537#issuecomment-294302961">here</a>.

>2. Setup.py (build & install) is completely redundant and it litters shit all over the f***ing place... At least include a decent uninstall option if that's what's gonna happen. This is mainly why I get so frustrated with this program. Again, no offense to anyone.

We are using the standard way for building, distributing, and installing modules using Distutils. I'm not sure what is the problem here. 

>3. Wifiphisher should be able to work 'right out of the box', no build or installation needed, just run it, no matter where it's run from. That's a really easy thing to do, just set a variable with the current working directory on each session!

This is not possible. Due to the complexity of our features we need to compile native code.

>4. Extra phishing pages should be able to be loaded externally so you don't have to copy files every time.

We haven't seen other people asking for it, but this could be a feature request.

I  believe you have posted on this forum under different handles in the past. This is not going to make your opinion stronger.

**Comment by yungtravla** [comment]
@sophron several pages in, and, the material titled 'The Cathedral and the Bazaar' and this Wifiphisher project seem to have little relevance to each another. 

Thanks for regarding all my points.

> Long story short, instead of complaining about the software, help make it better (a good way is by writing code).

I have in the past, and I would in the future if it became more modular, as far as I know, this entire issues page is meant for problems and suggestions.

> I think you are confusing logs with the output on the screen. Read (again?) my answer here.

I've seen more than just that request for a proper logger, and I would like this clear this up as well: logging basically means noting what happens, and wifiphisher does that, but it's like while wearing one of these:
<img width="200px" src="http://media.istockphoto.com/photos/slatted-sunglasses-picture-id158947005?k=6&m=158947005&s=612x612&w=0&h=If4EWOvsgO3x1hJN5IyHKdozWASogd2KpEq7hpcTE5E=">

> We haven't seen other people asking for it, but this could be a feature request.

It is a feature request.

What do you mean make my opinion stronger? This is the issues page for this tool, and I posted my issues with it. You're right about the better initiate being to write code instead of posting here, but don't start taking this the wrong way.

Rather than telling me whether things have been asked before or not, what logging and StdOut is, see these as suggestions to make this tool a lot more useful and user friendly.

**Comment by minanagehsalalma** [comment]
@anakin1028  there must be a  column for the entered passwords .

**Comment by blackHatMonkey** [comment]
@yungtravla the logging feature #747 is in progress and should be done soon.

**Comment by blackHatMonkey** [comment]
@yungtravla #747 has been implemented and now you can use the `--log-file` option to log information.
